fplus-http-server/src/middleware/signature_reader.rs

```diff
@@ -0,0 +1,61 @@
+use alloy::primitives::Address;
+use alloy::signers::Signature;
+use anyhow::Result;
+
+pub struct SignatureReader;
+
+impl SignatureReader {
+    pub async fn get_address_from_signature(
+        &self,
+        signature: &Signature,
+        message: &[u8],
+    ) -> Result<Address, alloy::signers::Error> {
+        let address_from_signature = signature.recover_address_from_msg(&message[..]).unwrap();
```

Comment:
Unwrap will probably mess with error handling if the signature is completely invalid.
```suggestion
        let address_from_signature = signature.recover_address_from_msg(&message[..])?;
```